File: leetcode/go/492.construct-the-rectangle.go
func constructRectangle(area int) []int {
    w := int(math.Floor(math.Sqrt(float64(area))))
    l := area / w
    for (w * l != area) {
        w -= 1
        l = area/w
    }
    return []int{l, w}
}

File: leetcode/rust-leetcode/src/492.construct-the-rectangle.rs
impl Solution {
    pub fn construct_rectangle(area: i32) -> Vec<i32> {
        let mut w = f64::sqrt(area as f64) as i32;
        let mut l = area / w;
        while w * l != area {
            w -= 1;
            l = area / w;
        }
        vec![l, w]
    }
}

